server/src/database/database-service.ts

```
@@ -0,0 +1,250 @@
+    );
+    await newDatabaseService.setTransactionId();
+    return newDatabaseService;
+  }
+
+  wrapQueryForTransaction = <
+    T extends { then: PromiseLike<any>["then"] } & Transactable
+  >(
+    query: T
+  ): T => {
+    const originalThen = query.then.bind(query);
+    query.then = (...args: any[]) =>
```

Biome (1.9.4)

[error] Do not add then to an object.



(lint/suspicious/noThenProperty)